### PR TITLE
From OpenLedger team 

### PR DESCRIPTION
**Pull request by alexandrty**
Implemented our coins in gate;
Added automatic calculation of the amount of the fee for withdrawing currency;
Remove currency wich have small fee pool balance;
User can choose fee currency in withdrawall page

### CODE REVIEW

**Comment by abitmore** [comment]
Please discuss in #232

**Comment by svk31** [comment]
I appreciate the PR, but there's a fair amount of bugs in this, for example the fees do not update when switching fee assets, and BTS is not included as a fee option for some reason. There are also styling issues with the modal fields non non-translated text.

I'll merge it to a branch then make some additional adjustments, it's a good opportunity for me to implement some fee helper classes I've been meaning to make for a while now.

File: web/app/components/Dashboard/SimpleDepositWithdraw.jsx [hunk 1]
```diff
@@ -145,9 +152,16 @@ class DepositWithdrawContent extends React.Component {
     }
 
     _updateAmount(amount) {
+        let fee = this._getFee();
+        let gateFee = parseFloat(this.props.gateFee);
+        let feeToSubtract = this.state.to_withdraw.asset_id !== fee.asset ? 0 : fee.amount;
+        let fee_precision = this.state.to_withdraw.precision;
         this.state.to_withdraw.setAmount({sats: amount});
+
+        let total_minus_fee = this.state.to_withdraw.getAmount({real: true}) - gateFee - (feeToSubtract/Math.pow(10,fee_precision))*1.095; //@#>
```
Comment (svk31) [inline web/app/components/Dashboard/SimpleDepositWithdraw.jsx]: Where does the `1.095` come from?

File: web/app/components/DepositWithdraw/blocktrades/WithdrawModalBlocktrades.jsx [hunk 2]
```diff
@@ -198,37 +228,84 @@ class WithdrawModalBlocktrades extends React.Component {
...
+            let asset = ChainStore.getObject(key);
+            let balanceObject = account_balances[key]?ChainStore.getObject(account_balances[key]):null;
+
+            if (balanceObject && balanceObject.get("balance") > 0) {
+                if(fee_asset_types.indexOf(key)==-1){
+                    asset_types.push(key);
+                }
+            }
+
+            if(asset&&utils.isValidPrice(asset.getIn(["options", "core_exchange_rate"]))&&parseInt(asset.getIn(["dynamic", "fee_pool"]), 10)>(this._feeBTS||2200)){
```
Comment (svk31) [inline web/app/components/DepositWithdraw/blocktrades/WithdrawModalBlocktrades.jsx]: Why 2200?

Comment (alexandrty) [inline web/app/components/DepositWithdraw/blocktrades/WithdrawModalBlocktrades.jsx]: Hello!
We realize that this solution requires further development, but it is needed to solve the problem asap taking into account public complaints.
Constants:
2200 - minimum balance for coin comission fee that is needed for coin display in the list of comission fee.
1.095 - we haven't completely solved this problem, but if the comission fee won't be multiplied by minimum coefficient (1,095),  the total sum will exceed the sum on the user balance and in fact the user won't be able to make funds withdrawal.

Comment (svk31) [inline web/app/components/DepositWithdraw/blocktrades/WithdrawModalBlocktrades.jsx]: Hm, not sure I understand. I don't think the gate fee should be subtracted there either, the gate fee just gets removed from the amount sent on the external blockchain, it has no impact on the amounts on the BTS side. The gate fee just needs to be displayed to the users so they know it's there.

I've reimplemented the fee calculations so they're much cleaner now, I think I'll merge this without the constants on our end and see if that causes any problems.

Comment (svk31) [inline web/app/components/DepositWithdraw/blocktrades/WithdrawModalBlocktrades.jsx]: So I guess the 1.095 just adds some margin in order to account for the extra fees added by the memo. I will leave it in for now and replace it once I have a proper memo fee estimation code added.

Comment (alexandrty) [inline web/app/components/DepositWithdraw/blocktrades/WithdrawModalBlocktrades.jsx]: Sigve, I fixed a bug with modal window https://github.com/alexandrty/bitshares-ui/commit/0a2c74b207049f5ab17a05ce5b56b96c8cbacaff

**Pull request merged**